cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/intervals/CassandraIntervalsWorker.java

```diff
@@ -0,0 +1,162 @@
+
+import com.datastax.oss.driver.api.core.MappedAsyncPagingIterable;
+import com.datastax.oss.driver.api.core.cql.AsyncResultSet;
+import com.datastax.oss.driver.api.core.cql.BoundStatementBuilder;
+import com.exactpro.cradle.cassandra.CassandraSemaphore;
+import com.exactpro.cradle.cassandra.dao.AsyncOperator;
+import com.exactpro.cradle.cassandra.iterators.IntervalsIteratorAdapter;
+import com.exactpro.cradle.intervals.Interval;
+import com.exactpro.cradle.intervals.IntervalsWorker;
+import com.exactpro.cradle.intervals.RecoveryState;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.function.Function;
+
+import static com.exactpro.cradle.cassandra.CassandraCradleStorage.TIMEZONE_OFFSET;
+
+public class CassandraIntervalsWorker implements IntervalsWorker
+{
+    private final CassandraSemaphore semaphore;
+    private final UUID instanceUuid;
+    private final Function<BoundStatementBuilder, BoundStatementBuilder> writeAttrs, readAttrs;
+    private final IntervalOperator intervalOperator;
+
+    public CassandraIntervalsWorker(CassandraSemaphore semaphore, UUID instanceUuid, Function<BoundStatementBuilder,
+            BoundStatementBuilder> writeAttrs, Function<BoundStatementBuilder,
+            BoundStatementBuilder> readAttrs, IntervalOperator intervalOperator)
+    {
+        this.semaphore = semaphore;
+        this.instanceUuid = instanceUuid;
+        this.writeAttrs = writeAttrs;
+        this.readAttrs = readAttrs;
+        this.intervalOperator = intervalOperator;
+    }
+
+    @Override
+    public void storeInterval(Interval interval) throws IOException
+    {
+        try
+        {
+            storeIntervalAsync(interval).get();
+        }
+        catch (Exception e)
+        {
+            throw new IOException("Error while storing interval "+interval.getId(), e);
+        }
+    }
+
+    @Override
+    public CompletableFuture<Void> storeIntervalAsync(Interval interval)
+    {
+        CompletableFuture<IntervalEntity> future = new AsyncOperator<IntervalEntity>(semaphore)
+                .getFuture(() -> {
+                    IntervalEntity intervalEntity = new IntervalEntity(interval, instanceUuid);
+                    return intervalOperator.writeInterval(intervalEntity, writeAttrs);
+                });
+
+        return future.thenAccept(r -> {});
+    }
+
+    @Override
+    public Iterable<Interval> getIntervals(Instant from, Instant to, String crawlerName, String crawlerVersion) throws IOException
+    {
+        try
+        {
+            return getIntervalsAsync(from, to, crawlerName, crawlerVersion).get();
+        }
+        catch (Exception e)
+        {
+            throw new IOException("Error while getting healing intervals from " + from, e);
```

Comment:
They are not healing intervals, they are just intervals.
And you're getting them "from" and "to". `crawlerName` would also be useful in the error.